test/helpers.lua

```diff
@@ -16,6 +16,9 @@ end
 local ok = function(res)
   return assert.is_true(res)
 end
+local substr = function(pattern, actual)
+  return assert.is_true(nil ~= string.find(actual, pattern, 1, true), "expected substring")
```

Comment:
substr typically returns a matched string. I'm not sure this function is actually needed.